packages/access/lib/constraints/requireExistingGrants.js

```diff
@@ -0,0 +1,51 @@
+const debug = require('debug')('access:lib:constraints:requireExistingGrants')
+
+/**
+ * Contraint checks if a granter has grants in campaign. If false, constraint
+ * will hinder display of campaign.
+ *
+ * Story: User should only see campaign if grants are in campaign. Grants may
+ * be pre-generated e.g. a series of voucher codes which may be printed on
+ * business cards.
+ *
+ * @example: {"requireExistingGrants": {}}
+ */
+
+const getMeta = async (args, context) => {
+  const { granter, campaign } = args
+  const { pgdb } = context
+
+  const hasGrants = await pgdb.query(`
+    SELECT "accessGrants".id
+
+    FROM "accessGrants"
+
+    WHERE
+      "accessGrants"."accessCampaignId" = '${campaign.id}'
+      AND "accessGrants"."granterUserId" = '${granter.id}'
+      AND "accessGrants"."revokedAt" IS NULL
+      AND "accessGrants"."invalidatedAt" IS NULL
+  `)
+
+  const meta = {
+    visible: hasGrants.length > 0,
```

Comment:
what does this mean for existing/future global campaings? Would they no tbe invisble to everyone initially?

Reply:
Constraints are bound per campaign via `campaigns.constraints`.

Some examples:

Campaign A's constraints:
```
[
  {
    "requireRole": {
      "roles": [
        "editor",
        "supporter",
        "admin"
      ]
    }
  },
  {
    "limitSlots": {
      "slots": 100
    }
  },
  {
    "limitRevokedSlots": {
      "slots": 100
    }
  },
  {
    "recipientInNoSlot": {}
  },
  {
    "recipientNotGranter": {}
  }
]
```

Campaign B's constraints:
```
[
  {
    "requireExistingGrants": {}
  },
  {
    "notGrantable": {}
  }
]
```

Further, by design `false` will trump `true` grantable and visible flags. So if there is one constraint returning `false` it won't be grantable or visible (aka. returned).